mls-rs/src/group/framing.rs

```diff
@@ -342,6 +342,45 @@ impl From<&PrivateMessage> for PrivateContentAAD {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub enum MlsMessageDescription<'a> {
+    Welcome {
+        key_package_refs: Vec<&'a KeyPackageRef>,
+        cipher_suite: CipherSuite,
+    },
+    ProtocolMessage {
+        group_id: &'a [u8],
+        epoch_id: u64,
+        content_type: ContentType, // commit, proposal, or application
+    },
+    GroupInfo,
+    KeyPackage,
+}
+
+impl MlsMessage {
+    pub fn description(&self) -> MlsMessageDescription<'_> {
+        match &self.payload {
+            MlsMessagePayload::Welcome(w) => MlsMessageDescription::Welcome {
+                key_package_refs: w.secrets.iter().map(|s| &s.new_member).collect(),
+                cipher_suite: w.cipher_suite,
+            },
+            MlsMessagePayload::Plain(p) => MlsMessageDescription::ProtocolMessage {
+                group_id: &p.content.group_id,
+                epoch_id: p.content.epoch,
+                content_type: p.content.content_type(),
+            },
+            #[cfg(feature = "private_message")]
+            MlsMessagePayload::Cipher(c) => MlsMessageDescription::ProtocolMessage {
+                group_id: &c.group_id,
+                epoch_id: c.epoch,
+                content_type: c.content_type,
+            },
+            MlsMessagePayload::GroupInfo(_) => MlsMessageDescription::GroupInfo,
+            MlsMessagePayload::KeyPackage(_) => MlsMessageDescription::KeyPackage,
```

Comment:
Shouldn't this give you access to the underlying key package data or group info data? I know that in 1.x we focused on what we need inside execution, but just for completeness it would be worth having more stuff here?

Reply:
It's not checking the signature though. I'm not sure it's safe to use?

Reply:
I think so, it's just a description, you can't really use what it tells you for anything bad because other API won't accept it

Reply:
You can't join a group or add somebody using it, but you may make decisions based on say custom group context extensions in group info. But the authenticity of that is not validated. I'm worried that this is a footgun for applications